### DISCUSSION TITLE
Sonoff D1 Dimmer Script

### DISCUSSION THREAD

**Discussion by Be2daErtin** [RavenSystem/esp-homekit-devices#2250] [MEPLHAA Scripts Collection]
Hi, 
It took some time but I manage to create a script for the D1 Dimmer

Device Name :iTead Sonoff D1 Dimmer
URL to Device: https://itead.cc/product/sonoff-d1-smart-dimmer-switch/
Used HAA version: 12.12.4
Summary of functions: Dimming Smart Switch
MEPLHAA script:
{ "c": { "io": [ [ [ 13 ], 2 ] ], "l": 13, "r": [ { "n": 0, "s": 9600, "p": 0, "b": 1 } ] }, "a": [ { "0": { "u": [ { "v": "aa550104000a0000ffffffffffffffff07" } ] }, "t": 30, "e": 5, "y0": [ { "0": { "u": [ { "v": "aa550104000aff00ffffffffffffffff06" } ] }, "v": 0 }, { "0": { "u": [ { "v": "aa550104000a0105ffffffffffffffff0d" } ] }, "v": 5 }, { "0": { "u": [ { "v": "aa550104000a010affffffffffffffff12" } ] }, "v": 10 }, { "0": { "u": [ { "v": "aa550104000a010fffffffffffffffff17" } ] }, "v": 15 }, { "0": { "u": [ { "v": "aa550104000a0114ffffffffffffffff1c" } ] }, "v": 20 }, { "0": { "u": [ { "v": "aa550104000a0119ffffffffffffffff21" } ] }, "v": 25 }, { "0": { "u": [ { "v": "aa550104000a011effffffffffffffff26" } ] }, "v": 30 }, { "0": { "u": [ { "v": "aa550104000a0123ffffffffffffffff2b" } ] }, "v": 35 }, { "0": { "u": [ { "v": "aa550104000a0128ffffffffffffffff30" } ] }, "v": 40 }, { "0": { "u": [ { "v": "aa550104000a012dffffffffffffffff35" } ] }, "v": 45 }, { "0": { "u": [ { "v": "aa550104000a0132ffffffffffffffff3a" } ] }, "v": 50 }, { "0": { "u": [ { "v": "aa550104000a0137ffffffffffffffff3f" } ] }, "v": 55 }, { "0": { "u": [ { "v": "aa550104000a013cffffffffffffffff44" } ] }, "v": 60 }, { "0": { "u": [ { "v": "aa550104000a0141ffffffffffffffff49" } ] }, "v": 65 }, { "0": { "u": [ { "v": "aa550104000a0146ffffffffffffffff4e" } ] }, "v": 70 }, { "0": { "u": [ { "v": "aa550104000a014bffffffffffffffff53" } ] }, "v": 75 }, { "0": { "u": [ { "v": "aa550104000a0150ffffffffffffffff58" } ] }, "v": 80 }, { "0": { "u": [ { "v": "aa550104000a0155ffffffffffffffff5d" } ] }, "v": 85 }, { "0": { "u": [ { "v": "aa550104000a015affffffffffffffff62" } ] }, "v": 90 }, { "0": { "u": [ { "v": "aa550104000a015fffffffffffffffff67" } ] }, "v": 95 }, { "0": { "u": [ { "v": "aa550104000a0164ffffffffffffffff6c" } ] }, "v": 100 } ] } ] }

Long description:
Sonoff D1 Dimmer script supports dimming function
Photos, additional information...:

**Comment by RavenSystem** [comment]
Great! Many thanks!
